### PR TITLE
e1s: init at 1.0.34

### PR DESCRIPTION
**Pull request by zelkourban**
## Description of changes
Add e1s
https://github.com/keidarcy/e1s

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/e1/e1s/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "e1s";
+  version = "1.0.34";
```
Comment (DontEatOreo) [inline pkgs/by-name/e1/e1s/package.nix:5]:
```suggestion
let
  pname = "e1s";
  version = "1.0.34";
in
buildGoModule {
```
Using `rec` is discouraged, you can learn more here: https://nix.dev/guides/best-practices#recursive-attribute-set-rec

File: pkgs/by-name/e1/e1s/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  #vendorHash = "sha256-fxo7TQjFrp2BbXz7bwMC5AuacKpyi5UmAMEVBmR91ko=";
+  vendorHash = "sha256-pM6win2W6HqjIg9QBblDl3ptKEI6sANliXkpwlFnTb0=";
+
+  meta = with lib; {
+    description = "Easily Manage AWS ECS Resources in Terminal 🐱";
+    homepage = "https://github.com/keidarcy/e1s";
+    changelog = "https://github.com/derailed/e1s/releases/tag/v${version}";
+    license = licenses.mit;
+    mainProgram = "e1s";
+    maintainers = with maintainers; [ keidarcy mdb ezafeire ];
```
Comment (DontEatOreo) [inline pkgs/by-name/e1/e1s/package.nix:22]:
```suggestion
  meta = {
    changelog = "https://github.com/derailed/e1s/releases/tag/v${version}";
    description = "Easily Manage AWS ECS Resources in Terminal 🐱";
    homepage = "https://github.com/keidarcy/e1s";
    license = lib.licenses.mit;
    mainProgram = "e1s";
    maintainers = with maintainers; [
      keidarcy
      mdb
      ezafeire
    ];
```

- Using with lib; is discouraged, you can learn more about it here https://nix.dev/guides/best-practices.html#with-scopes and here https://github.com/NixOS/nixpkgs/pull/293767
- Sorted alphabetically

File: pkgs/by-name/e1/e1s/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "keidarcy";
+    repo = "e1s";
+    rev = "v${version}";
+    hash = "sha256-fxo7TQjFrp2BbXz7bwMC5AuacKpyi5UmAMEVBmR91ko=";
+  };
+
+
+  #vendorHash = "sha256-fxo7TQjFrp2BbXz7bwMC5AuacKpyi5UmAMEVBmR91ko=";
+  vendorHash = "sha256-pM6win2W6HqjIg9QBblDl3ptKEI6sANliXkpwlFnTb0=";
```
Comment (DontEatOreo) [inline pkgs/by-name/e1/e1s/package.nix]: Is there any reason for the commented hash? If so, please add more context in the comment. I think you might have forgotten to remove it.

File: pkgs/by-name/e1/e1s/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "e1s";
+  version = "1.0.34";
+
+  src = fetchFromGitHub {
+    owner = "keidarcy";
+    repo = "e1s";
+    rev = "v${version}";
```
Comment (DontEatOreo) [inline pkgs/by-name/e1/e1s/package.nix:10]:
```suggestion
    rev = "refs/tags/v${version}";
```
This is to make sure we're getting the tag and not the branch.

File: pkgs/by-name/e1/e1s/package.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, lib, buildGoModule, fetchFromGitHub }:
```
Comment (DontEatOreo) [inline pkgs/by-name/e1/e1s/package.nix:1]:
```suggestion
{
  stdenv,
  lib,
  buildGoModule,
  fetchFromGitHub,
}:
```

Formatted as per RFC 0166

**Review by 4evy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/e1/e1s/package.nix (hunk 2 above)
Comment (DontEatOreo) [reply pkgs/by-name/e1/e1s/package.nix:22]: Just a note, none of these names are on the `maintainers/maintainer-list.nix`. Please read https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md on how to add yourself as a maintainer

File: pkgs/by-name/e1/e1s/package.nix (hunk 3 above)
Comment (zelkourban) [reply pkgs/by-name/e1/e1s/package.nix]: Yes, this was left by mistake, removed that in the later commit.

File: pkgs/by-name/e1/e1s/package.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+let
+  pname = "e1s";
+  version = "1.0.34";
+in
+buildGoModule {
```
Comment (eclairevoyant) [inline pkgs/by-name/e1/e1s/package.nix:6]:
```suggestion
buildGoModule {
  inherit pname version;
```

**Review by eclairevoyant — DISMISSED** [review]
I've one piece of feedback to fix eval:

File: pkgs/by-name/e1/e1s/package.nix (hunk 5 above)
Comment (eclairevoyant) [reply pkgs/by-name/e1/e1s/package.nix:1]: This isn't enforced as of yet, and once it's enforced would be done so by CI. Until then we don't need to nit about it.

But yes, eventually we will switch to using `nixfmt-rfc-style` treewide.

**Review by eclairevoyant — COMMENTED** [review]
Please rebase/squash into two commits, one for adding yourself to maintianer-list, and the other for the new package

**Comment by DontEatOreo** [comment]
If possible, please use your GitHub email when making the commits. If you don't want to share your GitHub email publicly, you can use the `noreply` email provided by GitHub for your account here: https://github.com/settings/emails

**Comment by DontEatOreo** [comment]
Please squash 891c1767a1a02b7e8dcc9532a3b4e7a08424e036 and either remove the commit message or include only the co-authors. Also, make a separate commit adding yourself as maintainer (`maintainers: add zelkourban`) and make sure that it is the first commit.

**Review by eclairevoyant — APPROVED** [review]

**Review by nicoonoclaste — APPROVED** [review]

**Pull request merged**